sklego/meta/_grouped_utils.py

```diff
@@ -61,8 +61,11 @@ def _split_groups_and_values(
     except (KeyError, IndexError):
         raise ValueError(f"Could not drop groups {groups} from columns of X")
 
+    check_X = kwargs.pop('check_X', True)
```

Comment:
I'm curious, why does `check_X` need to be removed from `kwargs`?

Reply:
Last time I checked, the methods under that line that take kwargs (`_check_grouping_columns` and/or `check_array`) complain about unknown parameter `check_X`. Thus my the `.pop`. However, I agree to follow the suggestion here https://github.com/koaning/scikit-lego/pull/495#discussion_r772919039